src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -132,30 +132,30 @@ func DeduceBrokers(
 			return []string{"127.0.0.1:9092"}
 		}
 
-		if len(conf.Redpanda.KafkaApi) == 0 {
+		if len(conf.Redpanda.AdvertisedKafkaApi) == 0 {
 			log.Trace(
-				"The config file contains no kafka listeners." +
-					" Empty redpanda.kafka_api.",
+				"The config file contains no advertised kafka listeners." +
+					" Empty redpanda.advertised_kafka_api.",
 			)
 			return []string{}
 		}
 
-		// Add the seed servers' Kafka addrs.
+		// Add the seed servers' advertised kafka addrs.
 		if len(conf.Redpanda.SeedServers) > 0 {
 			for _, b := range conf.Redpanda.SeedServers {
 				addr := fmt.Sprintf(
 					"%s:%d",
 					b.Host.Address,
-					conf.Redpanda.KafkaApi[0].Port,
+					conf.Redpanda.AdvertisedKafkaApi[0].Port,
```

Comment:
NIT: There could be helper function that iterate over advertised kafka api linsteners and create Permutation of seed server host address and available ports.

Reply:
I'm gonna do what we discussed a while back: https://github.com/vectorizedio/redpanda/pull/1334#issuecomment-832915118